db/document.go

```diff
@@ -1114,6 +1121,20 @@ func (doc *Document) UnmarshalWithXattrs(ctx context.Context, data, syncXattrDat
 				return pkgerrors.WithStack(base.RedactErrorf("Failed to unmarshal HLV during UnmarshalWithXattrs() doc with id: %s (DocUnmarshalAll/Sync).  Error: %v", base.UD(doc.ID), err))
 			}
 		}
+		if documentXattr != nil {
```

Comment:
I wonder if we can/should make this required. You'd have to implement this in rosmar but it feels like the code isn't going to behave correctly without this present.

Reply:
I'll leave this to be done as part of the rosmar ticket. I think I was just following pattern in the function currently for other xattrs.